src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
+        if ((tableNode = thisTable[index]) == null) {
+            thisTable[index] = newNode(hash(key), key, value);
+        } else {
+            Node<K, V> node;
+            if (tableNode.key != null && tableNode.key.equals(key) || tableNode.key == key) {
+                node = tableNode;
+            } else {
+                for ( ; ; ) {
+                    if ((node = tableNode.next) == null) {
+                        tableNode.next = newNode(hash(key), key, value);
+                        break;
+                    }
+                    if (node.key != null && node.key.equals(key) || node.key == key) {
+                        break;
+                    }
+                    tableNode = node;
+                }
+            }
+            if (node != null) {
+                node.value = value;
+                return;
+            }
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V>[] thisTable;
+        if ((thisTable = table) == null || thisTable.length == 0) {
+            return null;
+        }
+        int index = indexFor(hash(key),thisTable.length);
+        for (Node<K, V> node = thisTable[index]; node != null; node = node.next) {
+            if (node.key != null && node.key.equals(key) || node.key == key) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V>[] resize() {
+        Node<K, V>[] oldTable = table;
+        int oldCapacity = (oldTable == null) ? 0 : oldTable.length;
+        int oldThreshold = threshold;
+        int newCapacity = INITIAL_SIZE;
+        int newThreshold = (int) (LOAD_FACTOR * INITIAL_SIZE);
+        if (oldCapacity > 0) {
+            newCapacity = oldCapacity << 1;
+            newThreshold = oldThreshold << 1;
+        }
+        threshold = newThreshold;
+        Node<K, V>[] newTable = (Node<K, V>[]) new Node[newCapacity];
+        if (oldTable != null) {
+            transfer(newTable);
+        }
+        table = newTable;
+        return newTable;
+    }
+
+    private Node<K, V> newNode(int hash, K key, V value) {
+        return new Node<>(hash, key, value, null);
+    }
+
+    private void transfer(Node<K, V>[] newTable) {
+        Node<K, V>[] oldTable = table;
+        int oldCapacity = oldTable.length;
+        int newCapacity = newTable.length;
+        for (int i = 0; i < oldCapacity; ++i) {
+            Node<K, V> node = oldTable[i];
+            if (node != null) {
+                oldTable[i] = null;
+                do {
+                    Node<K,V> next = node.next;
+                    int index = indexFor(node.hash, newCapacity);
+                    node.next = newTable[index];
+                    newTable[index] = node;
+                    node = next;
+                } while (node != null);
```

Comment:
you can use rewritten method `put()` here instead of all this logic 
```suggestion
            Node<K, V> currentNode = oldTable[i];
                 while (currentNode != null) {
                     put(currentNode.key, currentNode.value);
                     currentNode = currentNode.next;
                 }
```